tests/test_config.py

```diff
@@ -796,3 +796,16 @@ def create_partial(model, optionals):
 
     # AssertionError: assert ['a', 'b'] == ['b']
     assert partial.model_json_schema()['required'] == ['b']
+
+
+def test_model_config_use_as_property():
+    """
+    Test: raises PydandicUserError when 'model_config' is used
+    as a property
+    """
```

Comment:
Though I appreciate the extra documentation, I think if you just rename this test to something like `test_model_config_as_model_field_raises`, the test speaks for itself :).

Reply:
I agree, thanks :)